azext_iot/tests/helpers.py

```diff
@@ -127,6 +128,233 @@ def create_storage_account(
+        "--queue-name {} --name {}".format(
+            namespace_name, rg, queue_name, policy_name
+        )
+    ).get_output_in_json()["primaryConnectionString"]
+
+    return queue_cs
+
+
+def create_cosmos_db(
+    cmd,
+    account_name: str,
+    database_name: str,
+    collection_name: str,
+    partition_key_path: str,
+    rg: str,
+    resource_name: str,
+    create_account: bool = True,
+    create_database: bool = True,
+    create_collection: bool = True,
+) -> str:
+    """
+    Create a cosmos db account, database, and collection (if needed) and return the cosmosdb connection string.
+    """
+
+    if create_account:
+        cmd(
+            'cosmosdb create --resource-group {} --name {} --tags iot_resource={}'.format(
+                rg, account_name, resource_name
+            )
+        )
+
+    if create_database:
+        cmd(
+            'cosmosdb sql database create --resource-group {} --account-name {} --name {}'.format(
+                rg, account_name, database_name
+            )
+        )
+
+    if create_collection:
+        cmd(
+            'cosmosdb sql container create --resource-group {} --account-name {} --database-name {} --name {} -p {}'.format(
+                rg, account_name, database_name, collection_name, partition_key_path
+            )
+        )
+
+    output = cmd(
+        'cosmosdb keys list --resource-group {} --name {} --type connection-strings'.format(rg, account_name)
+    ).get_output_in_json()
+
+    for cs_object in output["connectionStrings"]:
+        if cs_object["description"] == "Primary SQL Connection String":
+            return cs_object["connectionString"]
+
+
+def create_managed_identity(cmd, name: str, rg: str):
+    return cmd(
+        f"identity create -n {name} -g {rg}"
+    ).get_output_in_json()
+
+
+def get_role_assignments(cmd, scope, role):
+    return cmd(
+        'role assignment list --scope "{}" --role "{}"'.format(
+            scope, role
+        )
+    ).get_output_in_json()
+
+
+def assign_rbac_role(cmd, assignee: str, scope: str, role: str, max_tries: int = 10):
+    """
+    Assign a given role for a given assignee and scope. Will check for assignment completion.
+    """
+    tries = 0
+    while tries < max_tries:
+        role_assignments = get_role_assignments(cmd, scope, role)
+        role_assignment_principal_ids = [assignment["principalId"] for assignment in role_assignments]
+        if assignee in role_assignment_principal_ids:
+            break
+        # else assign role to scope and check again
+        cmd(
+            'role assignment create --assignee "{}" --role "{}" --scope "{}"'.format(
+                assignee, role, scope
+            )
+        )
+        sleep(10)
+        tries += 1
+
+
+def assign_cosmos_db_role(cmd, principal_id: str, role: str, cosmos_db_account: str, rg: str):
+    cmd(
+        "cosmosdb sql role assignment create -a {} -g {} --scope '/' -n '{}' -p {}".format(
```

Comment:
This is a blanket (everything) scope. Do we want to specify a more specific scope?

Reply:
blanket scope will make do for testing purposes